File: wallet_app/lib/src/feature/common/dialog/scan_with_wallet_dialog.dart
import 'package:flutter/material.dart';

import '../../../navigation/wallet_routes.dart';
import '../../../util/extension/build_context_extension.dart';
import '../../../util/extension/string_extension.dart';
import '../widget/text/body_text.dart';
import '../widget/text/title_text.dart';

class ScanWithWalletDialog extends StatelessWidget {
  const ScanWithWalletDialog({super.key});

  @override
  Widget build(BuildContext context) {
    return AlertDialog(
      title: TitleText(context.l10n.scanWithWalletDialogTitle),
      content: BodyText(context.l10n.scanWithWalletDialogBody),
      actions: <Widget>[
        TextButton(
          child: Text.rich(context.l10n.generalClose.toUpperCase().toTextSpan(context)),
          onPressed: () => Navigator.pop(context),
        ),
        TextButton(
          child: Text.rich(context.l10n.scanWithWalletDialogScanCta.toUpperCase().toTextSpan(context)),
          onPressed: () async {
            final navigator = Navigator.of(context);
            await navigator.pushNamedAndRemoveUntil(
              WalletRoutes.qrRoute,
              ModalRoute.withName(WalletRoutes.dashboardRoute),
            );
          },
        ),
      ],
    );
  }

  static Future<void> show(BuildContext context) {
    return showDialog<void>(
      context: context,
      builder: (BuildContext context) => const ScanWithWalletDialog(),
    );
  }
}

File: uiautomation/src/test/kotlin/nativescreen/disclosure/ScanWithWalletDialog.kt
package nativescreen.disclosure

import util.NativeMobileActions

class ScanWithWalletDialog : NativeMobileActions() {

    private val scanWithWalletDialogTitle = l10n.getString("scanWithWalletDialogTitle")
    private val scanWithWalletDialogBody = l10n.getString("scanWithWalletDialogBody")
    private val scanWithWalletDialogScanCta = l10n.getString("scanWithWalletDialogScanCta").uppercase()

    fun visible() = elementWithTextVisible(scanWithWalletDialogTitle)
    fun scanWithWalletDialogBodyVisible() = elementWithTextVisible(scanWithWalletDialogBody)
    fun scanWithWalletButtonVisible() = elementWithTextVisible(scanWithWalletDialogScanCta)
}
